docs/source/index.rst

```diff
@@ -85,3 +80,724 @@ Indices and Tables
+(stored at the location described by the environment variables
+``FOREST_CONFIG`` and ``QCS_CONFIG``, or else at the respective default
+locations ``~/.forest_config`` and ``~/.qcs_config``), which by default
+have the following respective contents:
+
+::
+
+    [Rigetti Forest]
+    url = https://api.rigetti.com/
+    key = None
+    user_id = None
+
+    [QPU]
+    exec_on_engage = :
+
+::
+
+    [Rigetti Forest]
+    qpu_endpoint_address = None
+    qvm_address = http://localhost:5000
+    compiler_server_address = http://localhost:6000
+
+These values control the following behaviors:
+
+-  ``Rigetti Forest``: This section contains network endpoint
+   information about the entire Rigetti Forest infrastructure, e.g.,
+   where to find information about which QPU devices are available.
+-  ``url``: This is the endpoint where pyQuil looks for device
+   information and for the 2.0 endpoints.
+-  ``key``: This stores the pre-2.0 API key.
+-  ``user_id``: This stores a 2.0 user ID.
+-  ``qpu_endpoint_address``: This is the endpoint where pyQuil will try to
+   communicate with the QPU orchestrating service during QPU-engagement.
+-  ``qvm_address``: This is the endpoint where pyQuil will try to
+   communicate with the Rigetti Quantum Virtual Machine. On a QMI, this
+   points to the provided QVM instance. On a local installation, this
+   should be set to the server endpoint for a locally running QVM
+   instance.
+-  ``compiler_server_address``: This is the endpoint where pyQuil will
+   try to communicate with the compiler server. On a QMI, this points to
+   a provided compiler server instance. On a local installation, this
+   should be set to the server endpoint for a locally running quilc
+   instance.
+-  ``QPU``: This section contains configuration information pertaining
+   to QPU access.
+-  ``exec_on_engage``: This is the shell command that the QMI will
+   launch when the QMI becomes QPU-engaged.
+
+    **NOTE:** PyQuil itself reads these values out using the helper
+    class ``pyquil._config.PyquilConfig``. PyQuil users should not ever
+    need to touch this class directly.
+
+
+Support
+~~~~~~~
+
+For support issues, please email ``support@rigetti.com``.
+
+
+Overview of updates to Quil and pyQuil
+--------------------------------------
+
+The primary differences in the programming language Quil 1.0 (as appearing in pyQuil 1.3) and Quil 2 (as appearing in
+2.0) amount to an enhanced memory model. Whereas the classical memory model in Quil 1.0 amounted to an flat bit array of
+indefinite size, the memory model in  Quil 2 is segmented into typed, sized, named regions.
+
+In terms of compatibility with Quil 1.0, this primarily changes how ``MEASURE`` instructions are formulated, since their
+classical address targets must be modified to fit the new framework. In terms of new functionality, this allows angle
+values to be read in from classical memory.
+
+Parametric programs
+^^^^^^^^^^^^^^^^^^^
```

Comment:
these carets (`^`) should be tildes (`~`) to match the other third-level headings